### ISSUE TITLE
Removing tabs doesn't free memory of each item from documents

### ISSUE THREAD

**Issue by Himanshu40** [BRL-CAD/arbalest#11]
If I create three tabs, then the size of the `documents` of type `unordered_map` is `3`. But closing three tabs doesn't free memory of each item from `documents`.

**Comment by Himanshu40** [comment]
It should be like the closing of tab should clear the corresponding `activeDocumentId` from documents of type `unordered_map`

**Comment by Himanshu40** [comment]
```cpp
MainWindow::~MainWindow()
{
    for (const std::pair<const int, Document *> pair: documents){
        Document * document = pair.second;
        delete document;
    }
}
```
Here `Document *` is being freed when the `MainWindow` is closed. I am thinking of a solution where for each tab is closed then the `document` corresponding to the tab should be deleted from memory.

**Comment by Himanshu40** [comment]
```cpp
const std::pair<const int, Document *> pair
```
Currently, `pair` is making copies each time iterating over `documents`. Instead of making copies use reference. #13 [2fecc35](https://github.com/BRL-CAD/arbalest/pull/13/commits/2fecc354bf7beba6373021b605016af8807d4961)

**Comment by Himanshu40** [comment]
Since `documents` is created in stack area so should leave it as it is or should erase each item from the `documents`?. `Document *` is being freed by the destructor that does its part.
```cpp
Document::~Document() {
    delete database;
}
```

**Comment by Himanshu40** [comment]
I have recorded on opening and closing of 5 tabs and the memory it occupies is 487 Mb and it's only being freed when I close the application. Btw is 487 Mb the max memory that the documents should be allocated? or does 5 documents size with the application GUI give 487 Mb? I think 487 Mb is the max size.

**Comment by Himanshu40** [comment]
Fixed #24
